### PR TITLE
Add PropTypes to card and button patterns

### LINKED ISSUE #16: Add PropTypes to components

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Added `defaultProps` and `propTypes` to existing components.

## QA

- Pull code
- Review code for sanity
- Run `npm run start`
- Navigate to [http://localhost:3000](http://localhost:3000)
- Check no console errors
- Run `npm run lint`
- Check no errors except the error outlined in #17

Fixes #16

### CODE REVIEW

File: src/components/Card/Card.js [hunk 1]
```diff
@@ -11,7 +12,22 @@ class Card extends React.Component {
...
+  imgSrc: 'http://placehold.it/200x32',
+  imgAlt: 'Placeholder',
+};
+
+Card.propTypes = {
+  modifier: PropTypes.string,
+  imgSrc: PropTypes.string,
+  imgAlt: PropTypes.string,
+  title: PropTypes.string.isRequired,
+  cardContent: PropTypes.string.isRequired,
```
Comment (Lukewh) [inline src/components/Card/Card.js]: If you change `props.cardContent` as I suggested before either here, or in another component, you can set this to `PropTypes.node`.

If you decide to use `{img: {src: '', alt: ''}}` as I also suggested you can do:
```
img: {
  src: PropTypes.string,
  img: PropTypes.string
}
```

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import './Button.css';
 
 class Button extends React.Component {
   render() {
     return (
-      <button className={`p-button--${this.props.type}`}>{this.props.value}</button>
-    )
+      <button className={`p-button--${this.props.modifier}`}>{this.props.value}</button>
```
Comment (anthonydillon) [inline src/components/Button/Button.js]: If there is no modifier does this return `p-button--`?

Comment (Lukewh) [reply src/components/Button/Button.js]: Good catch @anthonydillon, completely missed that

Comment (barrymcgee) [reply src/components/Button/Button.js]: `defaultProps` addresses this - if you check the `App.js`, base button has no modifier prop.

`Button.defaultProps = { modifier: 'base' };`

Comment (Lukewh) [reply src/components/Button/Button.js]: You're right @barrymcgee. That means I didn't miss it 😎

Comment (barrymcgee) [reply src/components/Button/Button.js]: Never doubted you @Lukewh 👊🏻

**Comment by barrymcgee** [comment]
@Lukewh I should have actioned all your suggestions here, would you mind casting another eye over? Many thanks :)

File: src/App.js [hunk 3]
```diff
@@ -6,12 +6,13 @@ import Button from './components/Button/Button';
 import Card from './components/Card/Card';
 
 class App extends Component {
-
-  cardImg = {
-    url : "http://placehold.it/200x32",
-    alt : "Placeholder"
+  constructor() {
+    super();
+    this.image = {
```
Comment (Lukewh) [inline src/App.js]: Sorry to say this now, only just thought of it, but if you're only going to use this 64x64 image for the cards, I'd move this from being an instance variable and just put it at the top of the `render` method as a normal const... But this works too, so up to you!

**Pull request merged**